habitat-lab/habitat/tasks/nav/nav.py

```diff
@@ -1200,9 +1200,14 @@ def __init__(self, *args: Any, **kwargs: Any):
         # Cache config
         self._lin_vel_range = self._config.lin_vel_range
         self._ang_vel_range = self._config.ang_vel_range
+        self._camera_ang_vel_range = self._config.camera_ang_vel_range
+        self._camera_pitch_ang_range = self._config.camera_pitch_ang_range
         self._enable_scale_convert = self._config.enable_scale_convert
         self._time_step = self._config.time_step
 
+        # Cache camera pitch angle
+        self.camera_pitch_ang = 0.0
```

Comment:
I have a feeling that the camera pitch state should be accessed through `self._sim`, instead of being a member variable within the Action object.
@ykarmesh What do you think?

Reply:
@exhaustin I was actually trying to get the camera pitch state from `self._sim`, but it seems that there is an issue of transformation and the robot node angle is not consistent. We also found that the output of angle() is not reliable when get the angle from the `self._sim` thank you!

Reply:
Can you elaborate more on what the issue of transformation is? It seems weird that you're able to correctly set the correct goal rigid state but unable to get the correct current rigid state.

Reply:
Yes, the angle read from the `self._sim` does not have a sign, maybe you could provide some insight how to get the sign as you did for getting the rotation of the robot?

Reply:
solved this issue! done!

Reply:
Sorry I didn't notice the comment, your solution looks good, thanks!